### ISSUE TITLE
Multilingual Content (de + en)

### ISSUE THREAD

**Issue by georgiee** [satellytes/satellytes.com#79]
+ We need to make the actual website multi lingual. 
+ The future blog doesn't need to translate blog posts.

We have spoken about this topic and we wanted to go back to contentful but another research revealed that gatsby has us covered since a while. Well at least a community plugin:
https://www.gatsbyjs.com/plugins/gatsby-plugin-intl/

With this we can create markdown content per language through a frontmatter property `lang`. Maybe that's easier than falling back to contentful which is less accessible then the github hosted content?

Let's discuss which way to go and then make our website duo-lingual (en + de).

**Comment by feedm3** [comment]
3 notes about this topic:
- speaking of default language: as the blog (probably) will be english, we should use english as default for all other sites
- i just remeber that dan abramov has a blog about react, which is also build on gatsby. his blog posts get translated to a lot of languages. maybe we can check how they handle things there and collect some inspiration: 
  - https://github.com/gaearon/overreacted.io
  - https://overreacted.io
- low prio: maybe we can try out a different cms: 
  - https://twitter.com/GatsbyJS/status/1391795288046776325: sanity is mentioned quite often
  - https://willit.build/

**Comment by georgiee** [comment]
overreacted uses https://github.com/angeloocana/gatsby-plugin-i18n
which is similar to https://www.gatsbyjs.com/plugins/gatsby-plugin-intl/

With our content being markdown based this should make it possible to have a file in the two formats:

+ `content.md`
+ `content.de.md`

The plugin will register adequate routes and uses react intl under the hood. Let's try this for our static content (index, career etc)

**Comment by georgiee** [comment]
I worked on this topic a little bit today. The current progress is not ready for a PR or draft PR (https://github.com/satellytes/satellytes.com/tree/gk-i18n).

+ I decided to go with [i18next](https://react.i18next.com/) as it has the much bigger ecosystem and it's more mature even though react-intl seems to be more popular on the first glance. The npm trend reveals they are similar popular.
+ I also installed [gatsby-plugin-i18n](https://github.com/angeloocana/gatsby-plugin-i18n) which is the route helper to serve different content. This will help us to serve the pages based on markdown content which are not meant to be translated by i18next
+ I created a npm task `extract-i18n` to use babel to extract the values to translate. It's a plain key/value json with the key being the translation key.
+ The translation files are always checked in (`locales/translation/de` etc).
+ The translation management can be done manually but I immediately created a project with my favorite translation tool [poeditor](poeditor.com]. I will create an account with our technical user to share it. It gives us the possibility to maintain our translations and to edit them easily. There are also great automations to quickly translate things. We need something like that because extracting will override the local files and this can get cumbersome. poeditor also has an API so we could send missing keys directly to poeditor. i18next offers some middleware but I don't know if there is one for poeditor existing.

Translation will work like that:
+ We can translate navigation items etc with i18n next
+ We can translate smaller text blocks with i18n next
+ Bigger chunks of text like our markdown files are not involved with i18n next and we will rely on the routing additions from `gatsby-plugin-i18n` to give us an extra page per markdown file. I have not touched this yet so I don't know if the two systems play well together because the i18n next gatsby plugin (to setup everything) also involves a little bit of routing.

That way we should be able to translate our current content 🙌
